front/src/components/store/ViewButton/ViewButton.stories.js

```diff
@@ -5,57 +5,49 @@ export default {
   component: ViewButton,
 }
 
-export const defaultStory = () => (
-  {
-    components: { ViewButton },
-    template: '<ViewButton />',
+const createViews = (nbr) => {
+  const data = []
+  for (let i = 1; i <= nbr; i++) {
+    data.push({
+      id: i,
+      text: `View ${i}`,
+      locked: i === nbr,
+    })
   }
-)
+  return data
+}
 
-defaultStory.storyName = 'default'
+const Template = (args, { argTypes }) => ({
+  components: { ViewButton },
+  props: Object.keys(argTypes),
+  template: '<ViewButton v-bind="$props" />',
+})
 
-export const overlayOpenedStory = () => (
-  {
-    components: { ViewButton },
-    template: '<ViewButton ref="vb" />',
-    mounted () {
-      this.$refs.vb.$el.querySelector('button').click()
-    },
-  }
-)
+export const DefaultStory = Template.bind({})
+DefaultStory.storyName = 'default'
 
-overlayOpenedStory.storyName = 'overlay opened'
-overlayOpenedStory.args = {
-  waitForSelector: '.p-overlaypanel',
-}
+const TemplateWithRef = (args, { argTypes }) => ({
+  components: { ViewButton },
+  props: Object.keys(argTypes),
+  template: '<ViewButton ref="vb" v-bind="$props" />',
+  mounted () {
+    this.$refs.vb.$el.querySelector('button').click()
+  },
+})
 
-export const overlayOpenedStoryWithViews = () => (
-  {
-    components: { ViewButton },
-    data () {
-      return {
-        views: [{
-          id: 1,
-          text: 'First view',
-        }, {
-          id: 2,
-          text: 'Second view',
-        }, {
-          id: 3,
-          text: 'Second view',
-          locked: true,
-        }],
-        value: '1',
-      }
-    },
-    template: '<ViewButton ref="vb" :views="views" v-model="value" />',
-    mounted () {
-      this.$refs.vb.$el.querySelector('button').click()
-    },
-  }
-)
+export const OverlayOpenedStory = TemplateWithRef.bind({})
+OverlayOpenedStory.storyName = 'overlay opened'
+
+export const OverlayOpenedStoryWithViews = TemplateWithRef.bind({})
+OverlayOpenedStoryWithViews.storyName = 'overlay opened with views'
+OverlayOpenedStoryWithViews.args = {
+  views: createViews(3),
+  value: '1',
+}
 
-overlayOpenedStoryWithViews.storyName = 'overlay opened with views'
-overlayOpenedStoryWithViews.args = {
-  waitForSelector: '.p-overlaypanel',
```

Comment:
I see you remove the `waitForSelector`, but imageshot seems "light",
Do you try with this special argument too, and were the imageshots the same ?
If so, that's ok for me to remove the `waitForSelector`.

Reply:
My bad, I didn't know it was useful

Reply:
Well, in fact it's a helper to tell puppeteer wait for this special element before taking screenshot.
I don't know if it "works" really, but that help us to always have the "better" imageshot before sending to CI.